wit/surface.wit

```diff
@@ -1,23 +1,24 @@
 package wasi:webgpu;
 
 interface surface {
-    use graphics-context.{graphics-context};
+    use graphics-context.{context};
     use wasi:io/poll@0.2.0.{pollable};
 
     record create-desc {
-        height: u32,
-        width: u32,
-        offscreen: bool,
+        height: option<u32>,
+        width: option<u32>,
     }
 
     resource surface {
         constructor(desc: create-desc);
 
-        connect-graphics-context: func(context: borrow<graphics-context>);
+        connect-graphics-context: func(context: borrow<context>);
 
         height: func() -> u32;
         width: func() -> u32;
 
+        try-set-size: func(height: option<u32>, width: option<u32>);
```

Comment:
Consider `request-set-size`? Try can imply throwing an exception if fails, although that is not as relevant in this context